apps/api/src/services/data-provider/coingecko/coingecko.service.ts

```diff
@@ -17,15 +17,25 @@ import {
   SymbolProfile
 } from '@prisma/client';
 import { format, fromUnixTime, getUnixTime } from 'date-fns';
-import got from 'got';
+import got, { Headers } from 'got';
 
 @Injectable()
 export class CoinGeckoService implements DataProviderInterface {
-  private readonly URL = 'https://api.coingecko.com/api/v3';
+  private headers: Headers = {};
+  private readonly URL: string = 'https://api.coingecko.com/api/v3';
 
   public constructor(
     private readonly configurationService: ConfigurationService
-  ) {}
+  ) {
+    if (this.configurationService.get('COINGECKO_API_KEY_DEMO')) {
```

Comment:
Maybe this is a bit cleaner:

```
const apiKeyDemo = this.configurationService.get('COINGECKO_API_KEY_DEMO');
const apiKeyPro = this.configurationService.get('COINGECKO_API_KEY_PRO');

if (apiKeyDemo) {
  this.headers['x-cg-demo-api-key'] = apiKeyDemo;
}

...
```

Reply:
Indeed, will be added in upcoming commit.